flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -1215,17 +1244,23 @@ private CompletableFuture<Void> waitForTerminatingJob(
                 getMainThreadExecutor());
     }
 
-    CompletableFuture<Void> getJobTerminationFuture(JobID jobId) {
+    private CompletableFuture<Void> getJobTerminationFutureOrFailedFutureForRunningJob(
```

Comment:
We could get rid of that change as well if we change the `thenComposeAsync` change in `Dispatcher#runJob`, I guess?